src/lib.rs

```diff
@@ -1574,14 +1620,150 @@ mod test {
         assert!(!_NEG1_2.is_integer());
     }
 
+    #[cfg(not(feature = "std"))]
+    use core::fmt;
+    #[cfg(not(feature = "std"))]
+    use fmt::Write;
```

Comment:
This needs to be `use core::fmt::Write;` for Rust 1.31 compatibility.

Or combine the two, `use core::fmt::{self, Write};`

Reply:
I managed to figure this out myself, but strangely, compiling locally gives a compile error (in `num-bigint`), whereas CI lets it pass. The error is
```
$ cargo build --no-default-features --features=bigint

Compiling num-bigint v0.3.0-pre (https://github.com/rust-num/num-bigint#a63ef963)  
error[E0658]: use of unstable library feature 'alloc': this library is unlikely to be stabilized in its current form or name (see issue #27783)
105 | extern crate alloc;
    | ^^^^^^^^^^^^^^^^^^^
```
This is using rust version `1.31.0`. `stable` does not have this issue.
```
$ rustup show
active toolchain
----------------
1.31.0-x86_64-unknown-linux-gnu (default)
rustc 1.31.0 (abe02cefd 2018-12-04)
```

Reply:
The `bigint` feature is now a no-std form, which requires Rust 1.36 for the `alloc` crate. Use `bigint-std` here to turn on full `num-bigint/std` in the dependency.

Reply:
When I ran `./ci/test_full.sh` on my local computer, it failed because I didn't have `TRAVIS_RUST_VERSION` set, so it used the wrong value for `NO_STD_FEATURES`. This problem is easy to work around for example with `TRAVIS_RUST_VERSION=1.31.0 ./ci/test_full.sh`, but this is hard to remember each time.

Do you think there is any value in falling back to querying `rustup show active-toolchain` in the case that `TRAVIS` is not equal to `true` ([which it always is in Travis](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables))? I realize it is a CI script, but it is very useful to run locally, to be able to tell if a change will break CI within 30 seconds locally as compared to 2-ish minutes on Travis (along with another `git push --force`).

Also I feel like the added (or removed) features are a bit of a minefield. The readme could better explain them (num-bigint does a good job). Maybe I'll add an issue/PR for this.

Reply:
> Do you think there is any value in falling back to querying `rustup show active-toolchain` in the case that `TRAVIS` is not equal to `true` ([which it always is in Travis](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables))?

I'd rather not assume `rustup` is present, but I think it could just use `rustc -V`.

A PR for feature docs would be great!